File: pubnub-java/src/main/java/com/pubnub/api/models/server/files/GeneratedUploadUrlResponse.java
package com.pubnub.api.models.server.files;

import com.google.gson.annotations.SerializedName;
import com.pubnub.api.models.consumer.files.PNUploadedFile;
import lombok.Data;

import java.util.List;

@Data
public class GeneratedUploadUrlResponse {
    private final Integer status;
    private final PNUploadedFile data;
    @SerializedName("file_upload_request")
    private final FileUploadRequest fileUploadRequest;

    @Data
    public static class FileUploadRequest {
        private final String url;
        private final String method;
        @SerializedName("expiration_date")
        private final String expirationDate;
        @SerializedName("form_fields")
        private final List<FormField> formFields;
    }
}

File: core/src/main/kotlin/com/pubnub/api/models/server/files/GeneratedUploadUrlResponse.kt
package com.pubnub.api.models.server.files

import com.google.gson.annotations.SerializedName
import com.pubnub.api.models.consumer.files.PNUploadedFile

data class GeneratedUploadUrlResponse(
    val status: Int,
    val data: PNUploadedFile,
    @SerializedName("file_upload_request")
    val fileUploadRequest: FileUploadRequest
) {

    data class FileUploadRequest(
        val url: String,
        val method: String,
        @SerializedName("expiration_date")
        val expirationDate: String,
        @SerializedName("form_fields")
        val formFields: List<FormField>
    )
}
